extracters/cleaners.py

```diff
@@ -22,26 +22,50 @@ def share_parse(value):
 
 def ymd_to_datetime(year, month, day, which="begin"):
 
-	# These might raise ValueError if the string is not an int
-	# which is fine at this stage
+
 	if type(year) is not int:
-		year = int(year)
+		try:
+			year = int(year)
+		except:
+			print("DATE CLEAN: year is %r" % year)
+			year = None
+
 	if type(month) is not int:
-		month = int(month)
+		try:
+			month = int(month)
+		except:
+			print("DATE CLEAN: month is %r" % month)
+			month = None
+
 	if type(day) is not int:
-		day = int(day)
+		try:
+			day = int(day)
+		except:
+			day = None
+
+	if not year:
+		if which == "begin":
+			year = 0
+		else:
+			year = 9999
 
-	if not month:
+	if not month or month > 12 or month < 0:
```

Comment:
Should probably be `month > 12 or month < 1` (?)

Reply:
if month == 0, it will already be caught by not month in the first part of the test.
But agree it reads strangely.